WaiterRobot/Core/Mvi/ObservableViewModel.swift

```diff
@@ -29,14 +29,73 @@ class ObservableViewModel<S: ViewModelState, E: ViewModelEffect, VM: AbstractVie
 
     @MainActor
     private func activate() {
-        guard task == nil else {
-            return
-        }
-        task = Task {
-            let logger = koin.logger(tag: VM.description())
-            for await state in actual.container.stateFlow {
-                self.state = state as! S
+        guard task == nil else { return }
+        task = Task { [weak self] in
+            guard let stateFlow = self?.actual.container.stateFlow else { return }
+
+            for await state in stateFlow {
+                self?.state = state as! State
             }
         }
     }
 }
+
+class TableListObservableViewModel: ObservableViewModel<TableListState, TableListEffect, TableListViewModel> {
```

Comment:
Very nice 🔥, but for me `ObservableTableListViewModel` would be more intuitive.
So that it has the same naming as in Kotlin, just with a prefix.